File: py/src/practise150/p14_trap_rain_test.py
from src.practise150.p14_trap_rain import Solution

def test_trap_rain():
    test_cases = [
        ([0,2,0,3,1,0,1,3,2,1] , 9),
        ([0,1,0,2,1,0,1,3,2,1,2,1], 6),
        ([4,2,0,3,2,5], 9)
    ]
    
    s = Solution()
    for test_case in test_cases:
        result = s.trap(test_case[0])
        assert result == test_case[1]

File: go/practise150/P14_trap_rain_test.go
package practise150

import (
	"fmt"
	"testing"
)

func TestTrapRain(t *testing.T) {
	type TestCase struct {
		input    []int
		expected int
	}
	test_cases := []TestCase{
		{
			input:    []int{0, 2, 0, 3, 1, 0, 1, 3, 2, 1},
			expected: 9,
		},
		{
			input:    []int{0, 1, 0, 2, 1, 0, 1, 3, 2, 1, 2, 1},
			expected: 6,
		},
		{
			input:    []int{4, 2, 0, 3, 2, 5},
			expected: 9,
		},
	}

	for idx, test_case := range test_cases {
		t.Run(fmt.Sprintf("Test_case_No:%d", idx+1), func(t *testing.T) {
			actual := TrapRain(test_case.input)
			if actual != test_case.expected {
				t.Errorf("Testing Trap Rain. Input: %v, Expected: %d, Actual: %d", test_case.input, test_case.expected, actual)
			}
		})
	}
}
